### ISSUE TITLE
Restore virtual server replacing main server's SSL certificate (with "when owner is different" option enable)

### ISSUE THREAD

**Issue by iliaross** [virtualmin/virtualmin-gpl#530]
The problem is related to virtual server restoration process, which is replacing main server's SSL certificate (which is a valid LE certificate) with self-signed (from the server that is being restored).

It's important to point out that I have SSL sharing enabled with option `Yes, even when owner is different`. The point is, that I have installed a default domain with wildcard certificate that actually covers all domains being used on the test system.

It works just great if you create a virtual server but if you restore it, then it just breaks the main domain certificate replacing it self-signed from restored domain. The restored domain was created on the different system with different SSL sharing options.

**Comment by jcameron** [comment]
Ok, so you are restoring a domain which is sharing it's SSL cert with another domain?

Check out the fix I just committed, which should resolve this ..

**Comment by iliajie** [comment]
Yes, this patches solves the initial problem! Thanks!

**Comment by jcameron** [comment]
I'm surprised I never noticed this before!

**Comment by jcameron** [comment]
In this case, is the domain the cert is shared with already on the destination system?

**Comment by iliajie** [comment]
> In this case, is the domain the cert is shared with already on the destination system?

No, because the restored domain doesn't exist on the destination system at the time of the restore. However, the main domain with wildcard Let's Encrypt certificate is present. Note that `nolink_certs` is set to `2`.

**Comment by jcameron** [comment]
Were the SSL file paths different between the old and new systems? Like one was using ~/ssl and the other was using /etc/ssl/virtualmin/id/ ?

**Comment by iliajie** [comment]
> Were the SSL file paths different between the old and new systems? Like one was using ~/ssl and the other was using /etc/ssl/virtualmin/id/ ?

No, the paths where the same!

**Comment by jcameron** [comment]
In that case, what paths were the restored domain's Apache config referring to?

**Comment by iliajie** [comment]
They were referring to the original paths existed on the source system, where SSL sharing is disabled.

**Comment by iliajie** [comment]
I have been testing today on a new dev systems with all latest patches applied, and oddly I could no longer reproduce the initial issue.

However, the different and related problem persists.

If **source** server where we make a backup has SSL sharing **enabled**, and **destination** system has SSL sharing **disabled**, then when a backup is restored it fails with an error .. Absolutely exactly the same error pops up if you try to later enable SSL website feature in _Edit Virtual Server_ page:

<img width="748" alt="image" src="https://user-images.githubusercontent.com/4426533/226114866-9d2c6562-b9c4-4e4b-a5a0-4c489b9c7ceb.png">

This issue should be directly related to how breaking of SSL sharing performed in a time of a restore. Because, if you break SSL sharing on the source system before making a backup then domain configs is getting cleared correctly (i.e. removing `ssl_combined` and `ssl_everything` entries with the self-signed certificate), however this doesn't happen upon restore the same correct way!

**Comment by jcameron** [comment]
Ok that seems like a different problem! Is this domain you're enabling SSL for an alias, or a regular domain?

**Comment by iliajie** [comment]
A regular top-level domain. As I mentioned in my previous comment this happens (presumably) because Virtualmin domain config happens to have incorrect (non existing) `ssl_combined` and `ssl_everything` records .. self-signed certificate not expected to have these files .. those records exist on original system because using legit LE SSL certificate, which _was shared_ with backed up server .. and it's breaking upon restore.

**Comment by jcameron** [comment]
Any chance I could login to your system on which you're seeing this issue? That error about `absolute_domain_path` in particular is confusing ..

**Comment by iliajie** [comment]
I could setup an access for you but it's a hustle as it's a local machine. However, you don't need my machine. I tried it on 3 separate installs (different distros) and the error persists the same on all.

Steps to reproduce it:

  1. Disable SSL certificate sharing in Virtualmin Configuration on target machine
   <img width="652" alt="image" src="https://user-images.githubusercontent.com/4426533/226194677-571b6ef2-450d-433b-952f-8960f05e2da2.png">

  3. Restore [this](https://github.com/virtualmin/virtualmin-gpl/files/11011739/ubuntu22-pro.local-ssl-sharing-enabled.tar.gz) backup

**Comment by jcameron** [comment]
Ok, trying that now ...

**Comment by jcameron** [comment]
Ok I just checked in a fix for this issue!

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
